overlord/snapstate/handlers.go

```diff
@@ -1942,6 +1942,115 @@ func (m *SnapManager) doPreferAliases(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+// nearlyReady returns whether all lanes (other than me) are Ready.
+func nearlyReady(me string, change *state.Change) bool {
+	for _, task := range change.Tasks() {
+		if me == task.ID() {
+			// ignore self
+			continue
+		}
+		if !task.Status().Ready() {
+			return false
+		}
+	}
+	return true
+}
+
+// laneSnapSetups returns the instance names from the first SnapSetup in each
+// non-zero lane that is Done
+func laneSnaps(change *state.Change) []string {
+	// change.Tasks() preserves the order tasks were added, otherwise it all falls apart
+	tasks := change.Tasks()
+	sadLanes := map[int]bool{}
+	for _, task := range tasks {
+		if task.Status() == state.DoneStatus {
+			continue
+		}
+		lanes := task.Lanes()
+		if len(lanes) != 1 {
+			// just ignore the task for now
+			continue
+		}
+		sadLanes[lanes[0]] = true
+	}
+
+	seenLanes := map[int]bool{}
+	var snapNames []string
+	for _, task := range change.Tasks() {
+		lanes := task.Lanes()
+		if len(lanes) != 1 || lanes[0] == 0 || sadLanes[lanes[0]] || seenLanes[lanes[0]] {
+			continue
+		}
+		var snapsup SnapSetup
+		if err := task.Get("snap-setup", &snapsup); err != nil {
+			continue
+		}
+		snapNames = append(snapNames, snapsup.InstanceName())
+		seenLanes[lanes[0]] = true
+	}
+
+	return snapNames
+}
+
+// reRefreshSetup holds the necessary details to re-refresh snaps that need it
+type reRefreshSetup struct {
+	UserID int `json:"user-id,omitempty"`
+	*Flags
+}
+
+var reRefreshUpdater = updateManyFromChange
```

Comment:
s/reRefreshUpdater/reRefreshUpdateMany/  , Updater makes it sound like it does, not just setup the update to me